soc/riscv/riscv-privileged/hpmicro/soc.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2022 HPMicro
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys/__assert.h>
+#include <hpm_common.h>
+#include <hpm_soc.h>
+#include <hpm_pmp_drv.h>
+#include <hpm_pllctl_drv.h>
+#ifdef CONFIG_XIP
+#include <hpm_bootheader.h>
+#endif
+
+#ifdef CONFIG_XIP
+/**
+ * @brief FLASH configuration option definitions:
+ * option[0]:
+ *    [31:16] 0xfcf9 - FLASH configuration option tag
+ *    [15:4]  0 - Reserved
+ *    [3:0]   option words (exclude option[0])
+ * option[1]:
```

Comment:
According to the description in the user manual:
this word should be option[0], not option[1]